### DISCUSSION TITLE
Sbt build throws The reference to entity "display" must end with the ';' delimiter.

### DISCUSSION THREAD

**Discussion by hugeband** [sbt/sbt#7212] [Q&A] [answered]
Hello, I have an SBT app with a Maven dependency, we built and released a version, but now that Renovate runs it is failing because of the next error, the maven dependency didn't change, renovate is for a third-party dependency, I tried to run local and it is not failing. Java apps that have the same dependency don't have issues. I created a dummy commit (adding a println) and while building it is throwing exceptions. All this is in drone running the image sbtscala/scala-sbt:eclipse-temurin-jammy-11.0.17_8_1.8.2_2.13.10, sbt version 18.2, scala version 2.13.10 

```scala
[error] org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 344; The reference to entity "display" must end with the ';' delimiter.
[error] 	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1243)
[error] 	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
[error] 	at lmcoursier.internal.shaded.coursier.core.compatibility.package$.xmlParseSax(package.scala:121)
[error] 	at lmcoursier.internal.shaded.coursier.maven.MavenRepository$.parseRawPomSax(MavenRepository.scala:75)
[error] 	at lmcoursier.internal.shaded.coursier.maven.MavenRepository.$anonfun$findVersioning$1(MavenRepository.scala:376)
[error] 	at lmcoursier.internal.shaded.coursier.util.EitherT.$anonfun$flatMap$1(EitherT.scala:18)
[error] 	at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$extension$1(Task.scala:14)
[error] 	at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$extension$1$adapted(Task.scala:14)
[error] 	at lmcoursier.internal.shaded.coursier.util.Task$.wrap(Task.scala:82)
[error] 	at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$2(Task.scala:14)
[error] 	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
[error] 	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
[error] 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] (update) org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 344; The reference to entity "display" must end with the ';' delimiter.
```

I appreciate if you have any idea how to fix the issue. Thanks

**Comment by eed3si9n** [comment]
It sounds like you have a corrupt POM file. Could you paste at least the file here so someone could have a hope of looking into this issue? Generally speaking without a way to reproduce on our computers, we can't give proper advice or work on a fix.

**Reply by hugeband** [reply, marked as answer]
I found the root cause, our project pulls from two repositories, one is only scala libraries, and another one is maven libraries. For some reason in the scala repository, we have a .pom file from a maven library but the content is HTML instead of XML, since the order of our repositories is first the scala and then the maven, sbt was pulling that .pom file and failing while parsing the HTML inside the file. We don't know why we have that file in the scala repository, our repository team is working on that.

**Reply by SethTisue** [reply]
perhaps the original cause of the bad file was #7202

**Reply by hygt** [reply]
Hello, we're facing the same issue at work and it indeed looks like corrupted cache due to the repo.scala-sbt.org outage.
It happens only when sbt uses our `sbt-plugins-release` mirror, and goes away if I make `.sbt/repositories` point to Maven central only.

We've asked our IT to wipe the entire thing but I was curious about the specific artifact resolution that would cause this error. However I couldn't manage to make sbt's bootstrap verbose enough to figure it out. I tried `--debug` and `--verbose` to no avail. What would be the proper approach in this case?
